horace_core/sqw/PixelData/@PixelTmpFileHandler/PixelTmpFileHandler.m

```diff
@@ -0,0 +1,117 @@
+classdef PixelTmpFileHandler
+
+properties (Constant, Access=private)
+    DIRTY_PIX_DIR_NAME_ = 'sqw_pix%05d';
+end
+
+properties (Access=private)
+    dirty_pix_dir_ = '';
+    pix_id_ = -1;
+end
+
+methods
+
+    function obj = PixelTmpFileHandler(pix_id)
+        % Construct a PixelTmpFileHandler object
+        %
+        % Input
+        % -----
+        % pix_id   The ID of the PixelData instance this object is linked to.
+        %          This allows this sets the tmp directory name.
+        obj.pix_id_ = pix_id;
+        dirty_pix_dir_name = sprintf(obj.DIRTY_PIX_DIR_NAME_, obj.pix_id_);
+        obj.dirty_pix_dir_ = fullfile(tempdir(), dirty_pix_dir_name);
+    end
+
+    function raw_pix = load_dirty_page(obj, page_number)
+        % Load a page of data from a tmp file
+        %
+        % Input
+        % -----
+        % page_number   The number of the page to read data from
+        %
+        tmp_file_path = obj.generate_dirty_pix_file_path_(page_number);
+        [file_id, err_msg] = fopen(tmp_file_path, 'rb');
+        if file_id < 0
+            error('PIXELTMPFIELHANDLER:load_dirty_page', ...
+                  'Could not open ''%s'' for reading:\n%s', tmp_file_path, ...
+                  err_msg);
+        end
+        try
+            raw_pix = fread(file_id, 'float32');
+        catch ME
+            fclose(file_id);
+            rethrow(ME);
+        end
+        fclose(file_id);
+    end
+
+    function obj = write_dirty_pix(obj, page_number, pix_data)
+        % Write the current page's pixels to a tmp file
+        %
+        % Inputs
+        % ------
+        % page_number   The number of the page being written, this set the tmp file name
+        % pix_data      The raw data to write
+        %
+        tmp_file_path = obj.generate_dirty_pix_file_path_(page_number);
+        if ~exist(obj.dirty_pix_dir_, 'dir')
+            mkdir(obj.dirty_pix_dir_);
+        end
+
+        file_id = fopen(tmp_file_path, 'wb');
+        if file_id < 0
+            error('PIXELTMPFIELHANDLER:write_dirty_pix', ...
+                  'Could not open file ''%s'' for writing.\n', tmp_file_path);
+        end
+
+        try
+            obj.write_pix_to_tmp_file_(file_id, pix_data);
+        catch ME
+            fclose(file_id);
```

Comment:
The `fclose(file_id)` can live in an `onCleanup` handler so you don't have to have it in the catch and in outside the block (the same pattern can be used in the `load_dirty_pix` method above